From 454a26821a50ad07850bd1361ca173a1ceec88b1 Mon Sep 17 00:00:00 2001
Message-Id: <454a26821a50ad07850bd1361ca173a1ceec88b1.1651138725.git.thomas@gllm.fr>
In-Reply-To: <dfbbc0d05a7d14b403668da8ce51c95172258a46.1651138725.git.thomas@gllm.fr>
References: <dfbbc0d05a7d14b403668da8ce51c95172258a46.1651138725.git.thomas@gllm.fr>
From: Pierre Ynard <linkfanel@yahoo.fr>
Date: Mon, 20 Jul 2020 13:11:34 +0200
Subject: [PATCH 07/18] stream_ReadLine: increase line length limit

Very long lines are occasionally encountered in text web resources such
as HTML, JSON or other API data. This bumps the hardcoded limit from
200 kB to 4 MB, which should be sufficient to support more than most of
them, but still reasonable to prevent any issue.

Fixes #24957

This is relevant for fixing YouTube playback in 3.0
Message-Id: <70866e5b1dbec781ad1c989bed9a40f37ddbb729.1599040437.git.nicolas@videolabs.io>
Message-Id: <c44db7efa3212b2974181646ba1c396cc6540fc4.1601374253.git.nicolas@videolabs.io>
Message-Id: <ba941d0b10888ffb960e66b73cbb80ddd9858af6.1602659942.git.nicolas@videolabs.io>
Message-Id: <e91c696392f6efcab00725488168e8bf2b82ae73.1605192544.git.nicolas@videolabs.io>
Message-Id: <31f8088b0a2c5080ec28dd611adbb5bde6ca63d0.1605701951.git.nicolas@videolabs.io>
Message-Id: <f7f93e77086f2cb8ffd53870de60d1a7a0e79244.1607001652.git.nicolas@videolabs.io>
Message-Id: <ab4ee97a6102316e29de2932653f17a72477bd4c.1628587500.git.thomas@gllm.fr>
---
 src/input/stream.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/input/stream.c b/src/input/stream.c
index b94279b431..6be87f8551 100644
--- a/src/input/stream.c
+++ b/src/input/stream.c
@@ -168,7 +168,7 @@ stream_t *(vlc_stream_NewMRL)(vlc_object_t* parent, const char* mrl )
  * \return A pointer to the allocated output string. You need to free this when you are done.
  */
 #define STREAM_PROBE_LINE 2048
-#define STREAM_LINE_MAX (2048*100)
+#define STREAM_LINE_MAX (2048*2048)
 char *vlc_stream_ReadLine( stream_t *s )
 {
     stream_priv_t *priv = (stream_priv_t *)s;
-- 
2.30.2

